### ISSUE TITLE
Negate assertion

### ISSUE THREAD

**Issue by tricoder42** [jfairbank/redux-saga-test-plan#65]
Hello and thank you for making this library! Great stuff! 🥇 

Is there a way to negate assertion? Sometimes I would like to test than an effect (`put`, `call`) isn't called under certain circumstances.

Suppose this trivial example:

```js
function* authRoot () {
  const token = yield select(authToken)
  if (token) yield call(setApiToken, token)
}
```

Now I can test, than `setApiToken` is really called with token:

```js
  it('should set Api auth token', function () {
    return expectSaga(authRoot)
      .withState({
        auth: {
          token: '42'
        }
      })
      .call(setApiToken, '42')
      .run()
  })
```

I would like to also test the `else` path and so that `setApiToken` isn't called when `token` is missing:

```js
  it('should skip missing token', function () {
    return expectSaga(authRoot)
      .withState({})
      .call(setApiToken, '42')  // this won't be called and it's OK
      .run()
  })
```

I haven't found any mention of this in docs. Is it supported?

Cheers

**Comment by jfairbank** [comment]
Hi, @tricoder42!

At the moment there isn't a way to test negations, but that would be a useful feature to have. I'll mark this thread as an enhancement. Thanks for the idea.

**Comment by nbotalov** [comment]
+1

**Comment by jfairbank** [comment]
@tricoder42 this is available now in v2.3.0 in `expectSaga`. You can use the `.not` property before an assertion to negate it now.

http://redux-saga-test-plan.jeremyfairbank.com/integration-testing/negated-assertions.html
